content/documentation/4-batch-developer-guides/2-batch/2-spring-batch.md

```diff
@@ -8,34 +8,310 @@ description: 'Create a Spring Batch Job'
+
+1.  Create a [BillingConfiguration](https://github.com/spring-cloud/spring-cloud-dataflow-samples/tree/master/dataflow-website/batch-developer-guides/batch/batchsamples/billrun/src/main/java/io/spring/billrun/configuration/BillingConfiguration.java) class in the `io.spring.billrun.configuration` using your favorite IDE that looks like the contents below.
+
+    ```java
+    {/* highlight-range{2-3} */}
+    @Configuration
+    @EnableTask
+    @EnableBatchProcessing
+    public class BillingConfiguration {
+      private static final Log logger = LogFactory.getLog(BillingConfiguration.class);
+
+      @Autowired
+      public JobBuilderFactory jobBuilderFactory;
+
+      @Autowired
+      public StepBuilderFactory stepBuilderFactory;
+
+      @Value("${usage.file.name:classpath:usageinfo.json}")
+      private Resource usageResource;
+
+      @Bean
+      public Job job1(ItemReader<Usage> reader,
+        ItemProcessor<Usage,Bill> itemProcessor, ItemWriter<Bill> writer) {
+          Step step = stepBuilderFactory.get("BillProcessing")
+                  .<Usage, Bill>chunk(1)
+                  .reader(reader)
+                  .processor(itemProcessor)
+                  .writer(writer)
+                  .build();
+
+          return jobBuilderFactory.get("BillJob")
+                  .incrementer(new RunIdIncrementer())
+                  .start(step)
+                  .build();
+      }
+
+      @Bean
+      public JsonItemReader<Usage> jsonItemReader() {
+
+          ObjectMapper objectMapper = new ObjectMapper();
+          JacksonJsonObjectReader<Usage> jsonObjectReader =
+                  new JacksonJsonObjectReader<>(Usage.class);
+          jsonObjectReader.setMapper(objectMapper);
+
+          return new JsonItemReaderBuilder<Usage>()
+                  .jsonObjectReader(jsonObjectReader)
+                  .resource(usageResource)
+                  .name("UsageJsonItemReader")
+                  .build();
+      }
+
+      @Bean
+      public ItemWriter<Bill> jdbcBillWriter(DataSource dataSource) {
+          JdbcBatchItemWriter<Bill> writer = new JdbcBatchItemWriterBuilder<Bill>()
+                          .beanMapped()
+                  .dataSource(dataSource)
+                  .sql("INSERT INTO BILL_STATEMENTS (id, first_name, " +
+                     "last_name, minutes, data_usage,bill_amount) VALUES " +
+                     "(:id, :firstName, :lastName, :minutes, :dataUsage, " +
+                     ":billAmount)")
+                  .build();
+          return writer;
+      }
+
+      @Bean
+      ItemProcessor<Usage, Bill> billProcessor() {
+          return new BillProcessor();
+      }
+    }
+    ```
+
+    The `@EnableBatchProcessing` annotation enables Spring Batch features and provide a base configuration for setting up batch jobs.
+    The `@EnableTask` annotation sets up a TaskRepository which stores information about the task execution such as the start and end time of the task and the exit code.
```

Comment:
I feel like we need more detail in this guide.  I'm not saying we need to turn people into Spring Batch experts, but calling out that the job has a single step that uses an `ItemReader`, an `ItemProcessor`, and an `ItemWriter` at a minimum should be done.  When you compare this to what we are using for inspiration (https://cloud.google.com/java/getting-started/hello-world for example), as I read through each section, I get a clear indication of what I'm looking at.  When I read this batch listing...as a batch newbie, I honestly don't know what I'm looking at.

Reply:
+ a whole bunch for thinking like a new developer.